src/sentry/integrations/discord/webhooks/command.py

```diff
@@ -68,46 +72,75 @@ def integration_spec(self) -> MessagingIntegrationSpec:
     @property
     def command_handlers(
         self,
-    ) -> Iterable[tuple[MessagingIntegrationCommand, Callable[[CommandInput], str]]]:
-        yield commands.HELP, self.help
-        yield commands.LINK_IDENTITY, self.link_user
-        yield commands.UNLINK_IDENTITY, self.unlink_user
-
-    def link_user(self, _: CommandInput) -> str:
-        if self.request.has_identity():
-            return ALREADY_LINKED_MESSAGE.format(email=self.request.get_identity_str())
-
-        if not self.request.integration or not self.request.user_id:
-            logger.warning(
-                "discord.interaction.command.missing.integration",
-                extra={
-                    "hasIntegration": bool(self.request.integration),
-                    "hasUserId": self.request.user_id,
-                },
+    ) -> Iterable[tuple[MessagingIntegrationCommand, CommandHandler[str]]]:
+        def help_handler(input: CommandInput) -> IntegrationResponse[str]:
+            return IntegrationResponse(
+                interaction_result=EventLifecycleOutcome.SUCCESS,
+                response=HELP_MESSAGE,
```

Comment:
Nit: I still think these are better left outside as their own methods, mainly because we can test them a bit more directly + for readability of this method. The `yield` flow is already a bit unique and stands alone as an Iterable pattern.

Reply:
To expand a little bit on this, I've been finding the type of high level testing we're doing (like `slack/webhooks/test_link_team` a bit tedious to get decent coverage on, while directly testing these methods would be much easier to validate behavior at a lower level. We can still test the top-level abstractions, and have the `ExampleIntegration` class/provider to help with this for more generic logic.